### PR TITLE
Minor Production trait cleanups.

### PR DESCRIPTION
**Pull request by pchote**
Fixes a handful of minor nits that were missed in #8979.

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/Production.cs [hunk 1]
```diff
@@ -118,17 +118,11 @@ public virtual bool Produce(Actor self, ActorInfo producee, string factionVarian
 			if (Reservable.IsReserved(self))
 				return false;
 
-			if (!occupiesSpace)
```
Comment (penev92) [inline OpenRA.Mods.Common/Traits/Production.cs]: I put this here for a reason. It is the `CanUseExit()` below that crashes, so moving this check won't work.

Comment (pchote) [inline OpenRA.Mods.Common/Traits/Production.cs]: See the other change in this patch, which prevents modders from defining exits on actors that don't occupy space.

**Comment by RoosterDragon** [comment]
:+1: Looks good to me.

**Comment by abcdefg30** [comment]
Looks also good to me. :+1:

**Pull request merged**